doc/website/release-notes/iceoryx-unreleased.md

```diff
@@ -362,6 +374,22 @@
     #include "iceoryx_hoofs/cxx/algorithm.hpp"
     constexpr uint32_t MAX_VAL = algorithm::maxVal(3, 1890, 57);
     constexpr uint32_t MIN_VAL = algorithm::minVal(3, 1890, 57);
+
+20. `mutex` must be always stored inside an `cxx::optional` and must use the builder pattern for
```

Comment:
This PR shows only an intermediate step. You are right and the ctor will become private so that this is not possible anymore. But this has to be done in a follow up PR otherwise the PR would explode with the mutex integration